openvdb_ax/openvdb_ax/codegen/VolumeComputeGenerator.cc

```diff
@@ -362,7 +362,8 @@ AttributeRegistry::Ptr VolumeComputeGenerator::generate(const ast::Tree& tree)
     // full code generation
     // errors can stop traversal, but dont always, so check the log
 
-    if (!this->traverse(&tree) || mLog.hasError()) return nullptr;
+    const size_t err = mLog.errors();
+    if (!this->traverse(&tree) || (mLog.errors() > err)) return nullptr;
```

Comment:
These changes to compare the value of errors() obviously rely on it being incrementable beyond maxErrors. If we want to keep errors() in line with the logged number,  we could wrap the mLog.error() calls on the compute generator and set some error bool member of ComputeGenerator or something? I feel like it is probably worth it to make it so that logger.errors() is the same as the logged ones? I know this isn't ideal and should get looked at again. But such a change would allow this to catch if a new error has been found in this generate call, even if an error existed on the logger beforehand? The use of the logger here was assuming a very simple linear workflow but thats not too robust.

Reply:
This also needs to be doen in Compile.cc in a bunch of methods too. I don't feel that the extra code/complexity necessary to solve this is worth it but I'll take another look.